api/v2_suppressions_test.go

```diff
@@ -0,0 +1,839 @@
+//
+// Author:: Darren Murray (<darren.murray@lacework.net>)
+// Copyright:: Copyright 2022, Lacework Inc.
+// License:: Apache License, Version 2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package api_test
+
+import (
+	"fmt"
+	"net/http"
+	"sort"
+	"testing"
+
+	"github.com/lacework/go-sdk/internal/lacework"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/lacework/go-sdk/api"
+)
+
+func TestSuppressionsAwsList(t *testing.T) {
+	fakeServer := lacework.MockServer()
+	fakeServer.UseApiV2()
+	fakeServer.MockToken("TOKEN")
+	fakeServer.MockAPI("suppressions/aws/allExceptions",
+		func(w http.ResponseWriter, r *http.Request) {
+			assert.Equal(t, "GET", r.Method, "AwsList() should be a GET method")
+			Suppressions := generateSuppressions()
+			fmt.Fprintf(w, Suppressions)
+		},
+	)
+	defer fakeServer.Close()
+
+	c, err := api.NewClient("test",
+		api.WithApiV2(),
+		api.WithToken("TOKEN"),
+		api.WithURL(fakeServer.URL()),
+	)
+	assert.NoError(t, err)
+
+	response, err := c.V2.Suppressions.Aws.List()
+	assert.Nil(t, err)
+	assert.NotNil(t, response)
+	var recsWithSuppressions []string
+	for key, sup := range response {
+		if len(sup.SuppressionConditions) >= 1 {
+			recsWithSuppressions = append(recsWithSuppressions, key)
+		}
+	}
+	assert.Equal(t, 5, len(recsWithSuppressions))
+	expectedRecsWithSup := []string{"AWS_CIS_1_14", "AWS_CIS_2_7", "AWS_CIS_4_4",
+		"LW_AWS_NETWORKING_50",
+		"LW_S3_1"}
+	sort.Strings(expectedRecsWithSup)
+	sort.Strings(recsWithSuppressions)
+	assert.Equal(t, expectedRecsWithSup, recsWithSuppressions)
+}
+
+func generateSuppressions() string {
+	return ListSuppressions()
```

Comment:
nit: Could collapse to one method here? I was looking for where ListSuppressions() was used - then I found this wrapper

Reply:
Fair point. I'll get that updated